### PR TITLE
WT-8849 in-memory RTS incorrectly rolls back "logged" files

### PR DESCRIPTION
**Pull request by keithbostic**
@kommiharibabu, @agorrod: This is the fix to revert the change in WT-8601 that changed in-memory databases so that timestamps applied to all objects created in an in-memory database, which broke the MDB build.

Here's the patch build: https://spruce.mongodb.com/version/6211a7bc3627e010721bec31/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC.

It looks clean to me but someone should please double-check that conclusion.

@lukech, this ticket is for [BF-24333](https://jira.mongodb.org/browse/BF-24333), so if we can get it into the merge soon, that would be terrific.

### CODE REVIEW

File: src/txn/txn_rollback_to_stable.c
Comment (kommiharibabu) [inline src/txn/txn_rollback_to_stable.c:1327]: space after `,`?

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/txn/txn_rollback_to_stable.c
Comment (keithbostic) [reply src/txn/txn_rollback_to_stable.c]: Fixed in e85715c.

**Comment by lukech** [comment]
Thanks for flagging it to me @keithbostic. Let's wait for @dgottlieb and @agorrod's lgtm before merging & vendoring it.

**Comment by dgottlieb** [comment]
I've added the in memory variant so we can verify the tests in https://jira.mongodb.org/browse/BF-24333 start passing. LGTM so long as they pass.

**Review by dgottlieb — APPROVED** [review]

**Review by lukech — APPROVED** [review]
I've checked the in-memory builder in the mongo patch build, there're no longer failure patterns as reported in BF-24333.

**Comment by agorrod** [comment]
Sorry about not actually LGTM'ing this - I was just closing out my tabs and saw that I never clicked the "approve" button :(

**Pull request merged**
